nhs/data/column_fix.py

```diff
@@ -0,0 +1,58 @@
+from typing import Iterable, Optional, Dict, Any, Tuple
+import os
+import polars as pl
+import pandas as pd
+from polars import LazyFrame
+
+from nhs.utils.path import list_files
+from nhs.utils.string import placeholder_matches
+from loguru import logger
+from .handling import read_spreadsheets
+
+
+"""
+Process the column names of a Pandas DataFrame to make them more readable.
+
+Args:
+    prev_csv_pl_df (dict[str, LazyFrame | None]): A dictionary containing Pandas DataFrames or None.
+    xlsx_pl_df: A Pandas DataFrame from an Excel file.
+
+Returns:
+    dict[str, LazyFrame | None]: A dictionary with updated column names based on the provided mapping.
+    
+Examples
+    --------
+    Assume we have the following files in the directory `"path/to/csv_files/"`:
+        - 2021Census_G01_AUS_AUS.csv
+        - 2021Census_G02_AUS_AUS.csv....
+    and "path/to/xlsx_files/":
+        - Metadata_2021_GCP_DataPack_R1_R2.xlsx
+        - 2021Census_geog_desc_1st_2nd_3rd_release.xlsx
+    Standard file in second sheet of Metadata_2021_GCP_DataPack_R1_R2.xlsx
+    "name of XLSX" : {"name of sheet1" : <dataframe>, "name of sheet2": <dataframe> ...}
+    By using read_spreadsheets to get dictionary containing dataFrames.
+    column_readable(
+        read_spreadsheets("path/to/csv_files/", 'csv'), 
+        read_spreadsheets("path/to/xlsx_files/", 'xlsx')
+        ) -> dict[str, LazyFrame | None]:
+"""
+
+
+def column_readable(prev_csv_pl_df: dict[str, LazyFrame | None], xlsx_pl_df) -> dict[str, LazyFrame | None]:
+    df = xlsx_pl_df['Metadata_2021_GCP_DataPack_R1_R2.xlsx']['Cell Descriptors Information']
+
+    result_dict = {}
+
+    for row in df.iter_rows(named=True):
+        key = row["DataPackfile"]
+        short = row["Short"]
+        long = row["Long"]
+        result_dict.setdefault(key, {})[short] = long
+
+    for key in prev_csv_pl_df:
+        try:
+            value = prev_csv_pl_df[key].rename(result_dict[key.split('_')[1]])
+            prev_csv_pl_df[key] = value
+        except KeyError:
+            continue
+    return prev_csv_pl_df

```

Comment:
So the first for-loop produces a dictionary of {` "G01" : { "short1": "long1", "short2": "long2"...}` ? You can try `df.group_by("DataPackfile").agg(pl.col("Short", "Long"))`, this will give you something like

```
┌──────────────┬─────────────────────────────────┬─────────────────────────────────┐
│ DataPackfile ┆ Short                           ┆ Long                            │
│ ---          ┆ ---                             ┆ ---                             │
│ str          ┆ list[str]                       ┆ list[str]                       │
╞══════════════╪═════════════════════════════════╪═════════════════════════════════╡
│ G01          ┆ ["Tot_P_M", "Tot_P_P", … "Age_… ┆ ["Total_Persons_Males", "Total… │
│ G02          ┆ ["Tot_P_F", "Age_0_4_yr_F", "A… ┆ ["Total_Persons_Females", "Age… │
│ G03          ┆ ["Age_0_4_yr_M", "Age_5_14_yr_… ┆ ["Age_groups_0_4_years_Males",… │
└──────────────┴─────────────────────────────────┴─────────────────────────────────┘
```

then in the second for-loop you can do `for datapack_code in set(names["Datapackfile"]) | set(prev_csv_pl_df.keys())`. `set` will convert the list of strings to a `set` object, which you can use `|` which denotes `union` (i.e. get the code that's in BOTH sets). This removes the need for the try-except block.